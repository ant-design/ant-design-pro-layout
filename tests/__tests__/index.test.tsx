import { mount, render } from 'enzyme';

import React from 'react';
import BasicLayout, { BasicLayoutProps } from '../../src/BasicLayout';
import { waitForComponentToPaint } from './util';

describe('BasicLayout', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => ({
        matches: false,
        addListener() {},
        removeListener() {},
      })),
    });
  });

  it('🥩 base use', async () => {
    const html = render(<BasicLayout />);
    expect(html).toMatchSnapshot();
  });

  it('🥩 support loading', async () => {
    const wrapper = mount(<BasicLayout loading />);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🥩 do not render menu', async () => {
    const wrapper = mount(<BasicLayout menuRender={false} />);
    await waitForComponentToPaint(wrapper);
    const menu = wrapper.find('.ant-pro-sider-menu');
    expect(menu.exists()).toBe(false);
    expect(
      (
        wrapper.find('section.ant-layout section.ant-layout').props().style ||
        {}
      ).padding,
    ).toBe(undefined);
    wrapper.unmount();
  });

  it('🥩 support menuDateRender', async () => {
    const wrapper = mount(
      <BasicLayout
        menuDataRender={() =>
          [
            {
              path: '/home',
              name: '首页',
              locale: 'menu.home',
              children: [
                {
                  path: '/home/overview',
                  name: '概述',
                  hideInMenu: true,
                  exact: true,
                  locale: 'menu.home.overview',
                },
                {
                  path: '/home/search',
                  name: '搜索',
                  exact: true,
                  hideInMenu: true,
                  locale: 'menu.home.search',
                },
              ],
            },
            {
              path: '/data_hui',
              name: '汇总数据',

              locale: 'menu.data_hui',
              children: [
                {
                  collapsed: true,
                  menuName: '域买家维度交易',
                  name: '域买家维度交易',

                  children: [
                    {
                      id: 2,
                      isNavHome: '2',
                      itemId: '191020104',
                      itemName: '_交易_买家_月表',
                      tab: 'adm_rk_cr_tb_trd_byr_ms',
                      tabProj: 'alifin_odps_birisk',
                      name: '_交易_买家_月表',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trd_byr_ms&tableSchema=alifin_odps_birisk',
                    },
                    {
                      id: 3,
                      isNavHome: '3',
                      itemId: '191020104',
                      itemName: '_航旅交易_买家_日表',
                      tab: 'adm_rk_cr_tb_trv_byr_ds',
                      tabProj: 'alifin_odps_birisk',
                      name: '_航旅交易_买家_日表',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trv_byr_ds&tableSchema=alifin_odps_birisk',
                    },
                  ],
                },
                {
                  collapsed: true,
                  name: '域买家维度交易2',
                  children: [
                    {
                      id: 5,
                      isNavHome: '2',
                      itemId: '191020107',
                      itemName: '_交易_买家_月表',
                      tab: 'adm_rk_cr_tb_trd_byr_ms',
                      tabProj: 'alifin_odps_birisk',
                      name: '_交易_买家_月表',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trd_byr_ms&tableSchema=alifin_odps_birisk',
                    },
                    {
                      id: 6,
                      isNavHome: '3',
                      itemId: '191020108',
                      itemName: '_航旅交易_买家_日表',
                      tab: 'adm_rk_cr_tb_trv_byr_ds',
                      tabProj: 'alifin_odps_birisk',
                      name: '_航旅交易_买家_日表',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trv_byr_ds&tableSchema=alifin_odps_birisk',
                    },
                  ],
                },
                {
                  collapsed: true,
                  name: '域买家维度交易3',
                  children: [
                    {
                      id: 7,
                      isNavHome: '2',
                      itemId: '191020107',
                      itemName: '_交易_买家_月表2',
                      tab: 'adm_rk_cr_tb_trd_byr_ms',
                      tabProj: 'alifin_odps_birisk',
                      name: '_交易_买家_月表2',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trd_byr_ms&tableSchema=alifin_odps_birisk',
                    },
                    {
                      id: 8,
                      isNavHome: '3',
                      itemId: '191020108',
                      itemName: '_航旅交易_买家_日表3',
                      tab: 'adm_rk_cr_tb_trv_byr_ds',
                      tabProj: 'alifin_odps_birisk',
                      name: '_航旅交易_买家_日表3',
                      path:
                        '/data_hui?tableName=adm_rk_cr_tb_trv_byr_ds&tableSchema=alifin_odps_birisk',
                    },
                  ],
                },
              ],
            },
            {
              path: '/data_ming',
              name: '明细数据',
              locale: 'menu.data_ming',
            },
            {
              path: '/other',
              name: '其他',

              locale: 'menu.other',
              children: [
                {
                  path: '/other/upLoad',
                  name: 'odps同步导入',
                  exact: true,
                  locale: 'menu.other.upLoad',
                  hideInMenu: true,
                },
                {
                  path: '/other/upLoadMenu',
                  name: '菜单导入',
                  exact: true,
                  locale: 'menu.other.upLoadMenu',
                  hideInMenu: true,
                },
                {
                  path: '/other/homeEdit',
                  name: '概述编辑',
                  exact: true,
                  locale: 'menu.other.homeEdit',
                  hideInMenu: true,
                },
              ],
            },
          ] as any
        }
      />,
    );
    await waitForComponentToPaint(wrapper);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('🥩 do not render footer', async () => {
    const wrapper = mount(<BasicLayout footerRender={false} />);
    await waitForComponentToPaint(wrapper);
    const footer = wrapper.find('footer');
    expect(footer.exists()).toBe(false);
    wrapper.unmount();
  });

  it('🥩 use onLogoClick', async () => {
    const onLogoClick = jest.fn();
    const wrapper = mount(
      <BasicLayout
        logo={
          <div onClick={onLogoClick} id="test_log">
            Logo
          </div>
        }
      />,
    );
    await waitForComponentToPaint(wrapper);
    const logo = wrapper.find('#test_log');
    logo.simulate('click');
    expect(onLogoClick).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('🥩 render logo', async () => {
    const wrapper = mount(<BasicLayout logo={<div id="test_log">Logo</div>} />);
    await waitForComponentToPaint(wrapper);
    const logo = wrapper.find('#test_log');
    expect(logo.text()).toEqual('Logo');
    wrapper.unmount();
  });

  it('🥩 render logo by function', async () => {
    const wrapper = mount(
      <BasicLayout logo={() => <div id="test_log">Logo</div>} />,
    );
    await waitForComponentToPaint(wrapper);
    const logo = wrapper.find('#test_log');
    expect(logo.text()).toEqual('Logo');
    wrapper.unmount();
  });

  it('🥩 onCollapse', async () => {
    const onCollapse = jest.fn();
    const wrapper = mount(<BasicLayout onCollapse={onCollapse} />);
    await waitForComponentToPaint(wrapper);
    wrapper
      .find('.ant-pro-sider-collapsed-button')
      .map((item) => item && item.simulate('click'));
    expect(onCollapse).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('🥩 siderWidth default', async () => {
    const wrapper = mount(<BasicLayout />);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-pro-sider').get(1).props.width).toBe(208);
    wrapper.unmount();
  });

  it('🥩 siderWidth=160', async () => {
    const wrapper = mount(<BasicLayout siderWidth={160} />);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-pro-sider').get(1).props.width).toBe(160);
    wrapper.unmount();
  });

  it('🥩 do not render collapsed button', async () => {
    const wrapper = mount(<BasicLayout collapsedButtonRender={false} />);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-pro-sider-collapsed-button').exists()).toBe(
      false,
    );
    wrapper.unmount();
  });

  it('🥩 when renderMenu=false, do not render collapsed button', async () => {
    const wrapper = mount(<BasicLayout menuRender={false} />);
    await waitForComponentToPaint(wrapper);
    expect(wrapper.find('.ant-pro-sider-collapsed-button').exists()).toBe(
      false,
    );
    wrapper.unmount();
  });

  it('🥩 render customize collapsed button', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout
        collapsedButtonRender={(collapsed) => (
          <span id="customize_collapsed_button">{`${collapsed}`}</span>
        )}
      />,
    );
    await waitForComponentToPaint(wrapper);
    const dom = wrapper.find('#customize_collapsed_button');
    expect(dom.text()).toEqual('false');

    wrapper.setProps({
      collapsed: true,
    });
    expect(dom.text()).toEqual('true');
  });

  it('🥩 do not render menu header', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout menuHeaderRender={false} />,
    );
    await waitForComponentToPaint(wrapper);
    const dom = wrapper.find('#logo');

    expect(dom.exists()).toBe(false);
    wrapper.unmount();
  });

  it('🥩 customize render menu header', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout
        menuHeaderRender={(logo, title) => (
          <div id="customize_menu_header">
            {logo}
            {title}
            <div id="customize_menu_header_text">customize_menu_header</div>
          </div>
        )}
      />,
    );
    await waitForComponentToPaint(wrapper);

    const dom = wrapper.find('#customize_menu_header');
    expect(dom.exists()).toBe(true);

    expect(dom.find('#customize_menu_header_text').text()).toEqual(
      'customize_menu_header',
    );
    wrapper.unmount();
  });

  it('🥩 contentStyle should change dom', async () => {
    const wrapper = render(
      <BasicLayout
        contentStyle={{
          padding: 56,
        }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('🥩 support className', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout
        className="chenshuai2144"
        contentStyle={{
          padding: 56,
        }}
      />,
    );
    expect(wrapper.find('div.chenshuai2144').exists()).toBeTruthy();
    wrapper.unmount();
  });

  it('🥩 support links', async () => {
    const wrapper = mount<BasicLayoutProps>(<BasicLayout links={['name']} />);
    await waitForComponentToPaint(wrapper);
    const dom = wrapper.find('.ant-pro-sider-link');
    expect(dom.exists()).toBeTruthy();
    wrapper.unmount();
  });

  it('🥩 do no render links', async () => {
    const wrapper = mount<BasicLayoutProps>(<BasicLayout />);
    await waitForComponentToPaint(wrapper);
    const dom = wrapper.find('.ant-pro-sider-link');

    expect(dom.exists()).toBeFalsy();
    wrapper.unmount();
  });

  it('🥩 pure style', async () => {
    const wrapper = mount<BasicLayoutProps>(<BasicLayout pure />);
    await waitForComponentToPaint(wrapper);
    const menu = wrapper.find('.ant-pro-sider-menu');
    expect(menu.exists()).toBe(false);
    const dom = wrapper.find('.ant-pro-sider-link');
    expect(dom.exists()).toBeFalsy();
    wrapper.unmount();
  });

  it('🥩 set page title render', async () => {
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout
        pageTitleRender={(props, pageName, info) => {
          if (info) {
            return info.pageName;
          }
          return pageName || 'ant';
        }}
      />,
    );
    await waitForComponentToPaint(wrapper);
    const dom = wrapper.find('.ant-pro-sider-link');

    expect(dom.exists()).toBeFalsy();
    wrapper.unmount();
  });

  it('🥩 onPageChange', async () => {
    const onPageChange = jest.fn();
    const wrapper = mount<BasicLayoutProps>(
      <BasicLayout
        onPageChange={onPageChange}
        location={{
          pathname: '/',
        }}
      />,
    );

    await waitForComponentToPaint(wrapper);
    wrapper.setProps({
      location: {
        pathname: '/name',
      },
    });

    expect(onPageChange).toBeCalled();
    wrapper.unmount();
  });

  it('🥩 fixSider and collapsed should have different style', async () => {
    const wrapper = mount<BasicLayoutProps>(<BasicLayout collapsed />);
    await waitForComponentToPaint(wrapper);

    let dom = wrapper.find('.ant-pro-fixed-header');
    expect(dom.exists()).toBeFalsy();
    wrapper.setProps({
      fixedHeader: true,
    });
    await waitForComponentToPaint(wrapper);
    dom = wrapper.find('header.ant-pro-fixed-header');
    expect(dom.exists()).toBeTruthy();
    expect(dom.props()?.style?.width).toBe('calc(100% - 48px)');

    wrapper.setProps({
      fixedHeader: true,
      collapsed: false,
    });

    dom = wrapper.find('header.ant-pro-fixed-header');
    expect(dom.props()?.style?.width).toBe('calc(100% - 208px)');

    wrapper.setProps({
      fixedHeader: true,
      collapsed: false,
      siderWidth: 120,
    });

    dom = wrapper.find('header.ant-pro-fixed-header');
    expect(dom.props()?.style?.width).toBe('calc(100% - 120px)');

    wrapper.setProps({
      fixedHeader: true,
      collapsed: false,
      menuRender: false,
    });

    dom = wrapper.find('header.ant-pro-fixed-header');
    expect(dom.props()?.style?.width).toBe('100%');

    wrapper.setProps({
      fixedHeader: true,
      layout: 'top',
    });

    dom = wrapper.find('header.ant-pro-fixed-header');
    expect(dom.props()?.style?.width).toBe('100%');
  });
});

it('🥩 support disableAutoContentMinHeight', async () => {
  const wrapper = mount<BasicLayoutProps>(
    <BasicLayout
      disableAutoContentMinHeight
      location={{
        pathname: '/',
      }}
    />,
  );
  await waitForComponentToPaint(wrapper);
  expect(wrapper.render()).toMatchSnapshot();
});
