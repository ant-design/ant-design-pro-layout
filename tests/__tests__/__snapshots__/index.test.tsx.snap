// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout 🥩 base use 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height:100%"
  >
    <aside
      class="ant-pro-sider ant-pro-sider-layout-side ant-layout-sider ant-layout-sider-dark"
      style="overflow:hidden;flex:0 0 208px;max-width:208px;min-width:208px;width:208px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex:1;overflow-y:auto;overflow-x:hidden"
        >
          <ul
            class="ant-menu ant-pro-sider-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
            style="width:100%"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-pro-sider-link-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-pro-sider-collapsed-button ant-menu-item-only-child"
              role="menuitem"
              style="padding-left:16px"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header"
        style="padding:0;height:48px;line-height:48px;width:100%;z-index:9"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex:1"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-basicLayout-children-content-wrap"
        />
      </main>
    </section>
  </section>
</div>
`;

exports[`BasicLayout 🥩 contentStyle should change dom 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height:100%"
  >
    <aside
      class="ant-pro-sider ant-pro-sider-layout-side ant-layout-sider ant-layout-sider-dark"
      style="overflow:hidden;flex:0 0 208px;max-width:208px;min-width:208px;width:208px"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex:1;overflow-y:auto;overflow-x:hidden"
        >
          <ul
            class="ant-menu ant-pro-sider-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
            style="width:100%"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-pro-sider-link-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-pro-sider-collapsed-button ant-menu-item-only-child"
              role="menuitem"
              style="padding-left:16px"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position:relative"
    >
      <header
        class="ant-layout-header"
        style="padding:0;height:48px;line-height:48px;width:100%;z-index:9"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex:1"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
        style="padding:56px"
      >
        <div
          class="ant-pro-basicLayout-children-content-wrap"
        />
      </main>
    </section>
  </section>
</div>
`;

exports[`BasicLayout 🥩 support loading 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <aside
      class="ant-pro-sider ant-pro-sider-layout-side ant-layout-sider ant-layout-sider-dark"
      style="overflow: hidden; flex: 0 0 208px; max-width: 208px; min-width: 208px; width: 208px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex: 1; overflow-y: auto; overflow-x: hidden;"
        >
          <ul
            class="ant-menu ant-pro-sider-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
            style="width: 100%;"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-pro-sider-link-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-pro-sider-collapsed-button ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 9;"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex: 1;"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-basicLayout-children-content-wrap"
        >
          <div
            style="padding-top: 100px; text-align: center;"
          >
            <div
              class="ant-spin ant-spin-lg ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
        </div>
      </main>
    </section>
  </section>
</div>
`;

exports[`BasicLayout 🥩 support menuDateRender 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <aside
      class="ant-pro-sider ant-pro-sider-layout-side ant-layout-sider ant-layout-sider-dark"
      style="overflow: hidden; flex: 0 0 208px; max-width: 208px; min-width: 208px; width: 208px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex: 1; overflow-y: auto; overflow-x: hidden;"
        >
          <ul
            class="ant-menu ant-pro-sider-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
            style="width: 100%;"
          >
            <li
              class="ant-menu-item ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                class="antd-menu-item-title"
              >
                首页
              </span>
            </li>
            <li
              class="ant-menu-submenu ant-menu-submenu-inline"
              role="menuitem"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-menu-submenu-title"
                role="button"
                style="padding-left: 16px;"
                title="汇总数据"
              >
                汇总数据
                <i
                  class="ant-menu-submenu-arrow"
                />
              </div>
              <div />
            </li>
            <li
              class="ant-menu-item ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                class="antd-menu-item-title"
              >
                明细数据
              </span>
            </li>
            <li
              class="ant-menu-item ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                class="antd-menu-item-title"
              >
                其他
              </span>
            </li>
          </ul>
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-pro-sider-link-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-pro-sider-collapsed-button ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 9;"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex: 1;"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-basicLayout-children-content-wrap"
        />
      </main>
    </section>
  </section>
</div>
`;

exports[`🥩 support disableAutoContentMinHeight 1`] = `
<div
  class="ant-design-pro ant-pro-basicLayout screen-md"
>
  <section
    class="ant-layout ant-layout-has-sider"
    style="min-height: 100%;"
  >
    <aside
      class="ant-pro-sider ant-pro-sider-layout-side ant-layout-sider ant-layout-sider-dark"
      style="overflow: hidden; flex: 0 0 208px; max-width: 208px; min-width: 208px; width: 208px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="ant-pro-sider-logo"
          id="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              src="https://gw.alipayobjects.com/zos/antfincdn/PmY%24TNNDBI/logo.svg"
            />
            <h1>
              Ant Design Pro
            </h1>
          </a>
        </div>
        <div
          style="flex: 1; overflow-y: auto; overflow-x: hidden;"
        >
          <ul
            class="ant-menu ant-pro-sider-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
            style="width: 100%;"
          />
        </div>
        <div
          class="ant-pro-sider-links"
        >
          <ul
            class="ant-menu ant-pro-sider-link-menu ant-menu-dark ant-menu-root ant-menu-inline"
            role="menu"
          >
            <li
              class="ant-menu-item ant-pro-sider-collapsed-button ant-menu-item-only-child"
              role="menuitem"
              style="padding-left: 16px;"
            >
              <span
                aria-label="menu-fold"
                class="anticon anticon-menu-fold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="menu-fold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                  />
                </svg>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </aside>
    <section
      class="ant-layout"
      style="position: relative;"
    >
      <header
        class="ant-layout-header"
        style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 9;"
      >
        <div
          class="ant-pro-global-header ant-pro-global-header-layout-side"
        >
          <div
            style="flex: 1;"
          />
        </div>
      </header>
      <main
        class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
      >
        <div
          class="ant-pro-basicLayout-children-content-wrap"
        />
      </main>
    </section>
  </section>
</div>
`;
